server/src/main/java/org/elasticsearch/script/BooleanFieldScript.java

```diff
@@ -45,6 +46,14 @@ public final void runForDoc(int docId) {
         execute();
     }
 
+    public final void runForDoc(int docId, Consumer<Boolean> consumer) {
+        runForDoc(docId);
+        int count = trues + falses;
+        for (int i = 0; i < count; i++) {
+            consumer.accept(i >= falses);
```

Comment:
I think we need to emit `true` `trues` times and then `false` `falses` times here?  Otherwise we're collapsing things down to a single value where there may be multiple values in the source.

Reply:
I think that is what I am doing: count is trues + falses. I emit false until there are falses left, and then trues. I may be missing something though as I don't follow the collapsing concern you mentioned :)

Reply:
duh yes, I am clearly not reading properly today.  Maybe add a comment?

Reply:
Maybe it's because the code is cryptic. this is some simplification that IntelliJ suggested, I should go back to something more readable although longer.